File: HJ87 密码强度等级.cpp
#include <iostream>
#include <string>
using namespace std;
int main()
    {
      string a;
    while(cin>>a)
        {
        int score=0;
        if(a.size()>=8)
           score+=25;
         else if(a.size()>=5&&a.size()<=7)
            score+=10;
        else 
            score+=5;
        
        int big=0;
        int small=0;
        
        for(auto c:a)
            {
            if(c>='a'&&c<='z')
              small++;
            if(c>='A'&&c<='Z')
                    big++;
        }
        
        if(small+big==0)
            ;
        else if(small&&big)
            score+=20;
        else
            score+=10;
        
        int num=0;
        
        for(auto c:a)
            {
            if(c>='0'&&c<='9')
              num++;
        }
        if(num>1)
           score+=20;
        else if(num==1)
            score+=10;
       
        if(a.size()-num-big-small>1)
            score=score+25;
        else if(a.size()-num-big-small==1)
            score=score+10;
        
        if(num&&small&big&(a.size()-num-big-small))
            score=score+5;
        else if(num&&(small||big)&(a.size()-num-big-small))
              score=score+3;
        else if(num&&(small||big))
            score=score+1;
        
        
        if(score>=90)
          cout<<"VERY_SECURE"<<endl;
            else if(score>=80)
               cout<<"SECURE"<<endl;
         else if(score>=70)
             cout<<"VERY_STRONG"<<endl;
         else if(score>=60)
             cout<<"STRONG"<<endl;
         else if(score>=50)
             cout<<"AVERAGE"<<endl;
         else if(score>=25)
             cout<<"WEAK"<<endl;
         else
             cout<<"VERY_WEAK"<<endl;
        
        
        
        
        
        
        
        
        
        
    }
     return 0;
}

File: HJ87 密码强度等级.py
while True:
    try:
        str_data = input().strip()
        num, up_char, low_char, other, score = 0, 0, 0, 0, 0

        for data in str_data:
            if data.isdigit():
                num += 1
            elif data.isalpha():
                if data.lower() == data:
                    low_char += 1
                else:
                    up_char += 1
            else:
                other += 1
        if len(str_data) < 5:
            score += 5
        elif len(str_data) < 8:
            score += 10
        else:
            score += 25
        if up_char==0 and low_char==0:
            pass
        elif (up_char==0 and low_char!=0) or (up_char!=0 and low_char==0):
            score += 10
        else:
            score += 20
        if num == 0:
            pass
        elif num == 1:
            score += 10
        else:
            score += 20
        if other == 0:
            pass
        elif other == 1:
            score += 10
        else:
            score += 25
        if num != 0 and (up_char+low_char) != 0 and other==0:
            score += 2
        elif num != 0 and up_char != 0 and low_char != 0 and other!=0:
            score += 5
        elif num != 0 and (up_char+low_char) != 0 and other!=0:
            score += 3
        if score >=90:
            print('VERY_SECURE')
        elif score >=80:
            print('SECURE')
        elif score >= 70:
            print('VERY_STRONG')
        elif score >= 60:
            print('STRONG')
        elif score >= 50:
            print('AVERAGE')
        elif  score>=25:
            print('WEAK')
        else:
            print('VERY_WEAK')
    except:
        break
